### PR TITLE
Use random shuffle for active_transactions::request_confirm ()

### PR DESCRIPTION
**Pull request by SergiySW**
for blocks rebroadcast & confirmation request in overload mode

### CODE REVIEW

File: nano/node/node.cpp [hunk 1]
```diff
@@ -3617,16 +3608,30 @@ void nano::active_transactions::request_confirm (std::unique_lock<std::mutex> &
...
+		}
 		node.network.republish_block_batch (rebroadcast_bundle);
 	}
 	// Batch confirmation request
-	if (!nano::is_live_network && !requests_bundle.empty ())
+	if (!requests_bundle.empty ())
 	{
+		while (requests_bundle.size () > max_broadcast_queue)
+		{
+			requests_bundle.erase (requests_bundle.begin ());
```
Comment (wezrule) [inline nano/node/node.cpp]: you can probably use the ranged version of `erase` instead of a loop

File: nano/node/node.cpp [hunk 2]
```diff
@@ -3557,10 +3554,7 @@ void nano::active_transactions::request_confirm (std::unique_lock<std::mutex> &
 					// broadcast_confirm_req_base modifies reps, so we clone it once to avoid aliasing
 					if (!nano::is_test_network)
 					{
-						if (confirm_req_bundle.size () < max_broadcast_queue)
-						{
-							confirm_req_bundle.push_back (std::make_pair (i->election->status.winner, reps));
-						}
+						confirm_req_bundle.push_back (std::make_pair (i->election->status.winner, reps));
```
Comment (wezrule) [inline nano/node/node.cpp]: `emplace_back` can be used, which can save a copy/move, i.e:
`confirm_req_bundle.emplace_back (i->election->status.winner, reps);`

Comment (SergiySW) [reply nano/node/node.cpp]: Thanks!

Now I guess we better make proper ordering for active transactions instead of random

**Comment by SergiySW** [comment]
Closing in favor of https://github.com/nanocurrency/nano-node/pull/1810

**Pull request closed without merging**